srcStatic/ConsoleModuleLoader.cpp

```diff
@@ -58,27 +58,34 @@ int __fastcall GetArtPath(void*, int, char*, char*, char *destBuffer, int buffer
 }
 
 // Returns an empty string if no module is found or if the module request is ill-formed.
-std::string ConsoleModuleLoader::FindModuleDirectory()
+std::string ConsoleModuleLoader::ParseCommandLine()
 {
 	auto arguments = GetCommandLineArguments();
 
-	if (arguments.size() == 0) {
+	const std::string switchName = GetSwitch(arguments);
+
+	if (switchName.empty()) {
 		return std::string();
 	}
 
-	std::string switchName = arguments[0];
-	arguments.erase(arguments.begin()); //Remove switchName from arguments
-
-	if (!ParseArgumentName(switchName)) {
-		return std::string();
+	if (switchName == "loadmod") {
+		return ParseLoadModCommand(arguments);
 	}
 
-	if (switchName != "loadmod") {
-		PostErrorMessage("ConsoleModuleLoader.cpp", __LINE__, "Provided switch is not supported: " + switchName);
-		return std::string();
+	PostErrorMessage("ConsoleModuleLoader.cpp", __LINE__, "Provided switch is not supported: " + switchName);
```

Comment:
Yes, this error check now lives in ConsoleArgumentParser. To translate the error check to the dll code would reduce the amount of code available for testing so I think it would make sense to leave it in place at this new location.

I forgot to update the line "ConsoleModuleLoader.cpp" to be "ConsoleArgumentParser.cpp". Will fix.

Reply:
Maybe you can use the `__FILE__` macro instead of a hardcoded string. Usually I see `__FILE__` and `__LINE__` used together.

Reply:
Wish I had known about the __FILE__ macro a long time ago! Good call.